exercise/problem/requests.go

```diff
@@ -1,31 +1,41 @@
 package main
 
 import (
+	"context"
 	"io"
 	"net/http"
 	"os"
 	"sync"
 )
 
 func main() {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
It's a good practice (sometimes a must) to place a 

`defer cancel()`

below this line, in order to ensure the threads are cancelled always (even if the prorgram success). If there's no explicit cancellation by default the code could drive to a thread leak.